networkx/algorithms/dag.py

```diff
@@ -477,6 +507,36 @@ def is_aperiodic(G):
     NetworkXError
         If `G` is not directed
 
+    Examples
+    --------
+    A graph consisting of one cycle,
+    the length of which is 2
+    (k = 2 divides the length of every cycle in the graph
+    and therefore the graph is not aperiodic)::
+
+        >>> DG = nx.DiGraph([(1, 2), (2, 1)])
+        >>> nx.is_aperiodic(DG)
+        False
+
+    A graph consisting of two cycles,
+    the length of which differs by one
+    (cycle lengths are coprime,
+    so there is no k > 1 dividing each cycle length
+    and therefore the graph is aperiodic)::
+
+        >>> DG = nx.DiGraph([(1, 2), (2, 3), (3, 1), (1, 4), (4, 1)])
+        >>> nx.is_aperiodic(DG)
+        True
```

Comment:
It's certainly not necessary, but you could add an example here w/ a DiGraph that contains two cycles whose lengths are not the same but share a common factor, e.g. k=4 and k=2.